src/parserfunc.c

```diff
@@ -866,7 +866,7 @@ pf_do_not(ParseNode* self)
     {
         set_error(self->state, PARSER_ERR_OVERFLOW, NULL);
         free(ans);
-        ans = NULL;
+        return NULL;
```

Comment:
This leaks `val`.  I'd rather guard the `mp_not()` call below instead of returning early here.

Reply:
oh, yes, thanks for flooking over this.